### DISCUSSION TITLE
Transition Option

### DISCUSSION THREAD

**Discussion by BenjaminOddou** [DerYeger/yeger#63] [Ideas]
Hey 👋,

I am trying to implement transitions in my gallery. However, it seems laggy when there is a lot of elements. I just saw that there is a special component for transitioning lists of items (`<TransitionGroup>`). Maybe it can be incorporated as an option in the plugin ? 

[Documentation](https://vuejs.org/guide/built-ins/transition-group.html)

**Comment by DerYeger** [comment]
Thanks for the suggestion!

I implemented a prototype, but I don't think it makes sense to use a `TransitionGroup` with this package. Since both columns and items change on every change to the `items` prop, all animations play every time, regardless of the individual item.

**Comment by BenjaminOddou** [comment]
ok I understand. So there is no proper way to implement transition in this module ? Another question not related to the topic directly but is there a way to rerender the elements (redraw) when a button is clicked ?

**Reply by DerYeger** [reply]
I'm not sure, `TransitionGroup` doesn't seem to work. But I'm open for PRs, so please try it yourself!

Regarding the redraw on click, did you try recreating the `items` prop, i.e., `items = [...items]`?

**Reply by BenjaminOddou** [reply]
I am not sure I am doing it the right way. In my app, I am fetching data from my api returning an array of objects. When I want to add elements, I am refreshing the api call with new transform parameters. What I find weird, is that the elements are not always placed correctly when I refresh the api.

I found that I have this error now :
```sh
ResizeObserver loop completed with undelivered notifications.
```

see below:

```vue
<script setup lang="ts">
const imageNumber = ref(25) // max 25 images to return

const {
  data: images,
  pending,
  refresh
} = await useFetch<ImageKit[]>('/api/imgkit', {
  transform: (images) => {
    return images
      .filter((image) => image.filePath.includes('galerie-page')) // filtering images based on filePath property
      .sort((imageA, imageB) => (imageA.name < imageB.name ? 1 : -1)) // Sorting images based on name
      .slice(0, imageNumber.value) // Limiting the response object to a reactive number
  },
  initialCache: false
})

watch(imageNumber, () => {
  refresh() // When the reactive number change, refresh the API
})

const onRedraw = async () => {
  await nextTick()
  requestAnimationFrame(() => {
    requestAnimationFrame(() => {
      ScrollTrigger.refresh()
    })
  })
}
</script>

<template>
        <MasonryWall
          id="galerie-masonry"
          :items="images"
          :column-width="340"
          :gap="0"
          @redraw="onRedraw"
        >
          <template #default="{ item }: { item: ImageKit }">
              <nuxt-img
                v-show="item.tags && item.tags.find((tag) => tag === filter)"
                :src="item.filePath"
                width="700"
                class="pl-6 pr-6 pb-6 sm:pl-0"
              />
          </template>
        </MasonryWall>

          <button
            id="see-more"
            class="button flex w-full cursor-pointer justify-center p-2 disabled:pointer-events-none disabled:bg-light-lavender dark:disabled:pointer-events-none dark:disabled:bg-light-orange"
            @click="imageNumber = imageNumber + 25"
          >
            Show more
          </button>
</template>
```

**Reply by DerYeger** [reply]
You'll have to publish a reproduction if you need my help.

Did you try `@click="() => { imageNumber = imageNumber + 25" }`?

**Reply by BenjaminOddou** [reply]
Hey @DerYeger, the error message doesn't seems to be linked to my issues (apologies). This is only happening in Safari (even on your demo) 
![safari](https://user-images.githubusercontent.com/85166574/191092918-7dd3e246-9ff4-44de-ac5c-8c8c73c8667b.png)

Maybe this can help ? (https://github.com/juggle/resize-observer/issues/115)

I can't really replicate sample from my project for now. What I was looking for is advice about the way implement new items. I know that the common way to do it is to use spread operator to recreate the items prop but for my use case I am not sure how to implement it. 😅

Also this code is working I mean I can see new images coming but for some reason it happens sometimes that new images are not spread correctly between the different columns (like 10 images in 1 col). I tried to await the refresh but it still seems to 'bug' on some situations. As I don't really understand why, I'd like to know if there is a way for us to manually trigger the render function of the plugin.

Thank you for your patience and all your time you took to respond, I truly appreciate it 🙂

**Reply by DerYeger** [reply]
Again, I just cant help without a reproduction.

Last tip I can give is not refreshing your fetch call (since the fetched data doesn't change), and instead performing the `filter`, `sort`, and `slice` in a `computed`.
This would also make your `watch` redundant.

**Reply by BenjaminOddou** [reply]
Thank you so much @DerYeger, your last tip helped me to figure out my problem. It's way more fluid now without having to refresh the API 🙂.

**Reply by DerYeger** [reply]
> Thank you so much @DerYeger, your last tip helped me to figure out my problem. It's way more fluid now without having to refresh the API 🙂.

Awesome!
